docs/topics/native/native-spm.md

```diff
@@ -0,0 +1,192 @@
+  1. Only one consumer package in a multi-package project can depend on the external module, to avoid dependency conflicts
+  within the project. So all the logic depending on your KMP module should be encapsulated in a particular consumer package.
+  2. If you publish the KMP project using an automated CI process, this process would need to include publishing the
+  updated `Package.swift` file to the consumer repo. Such a phase in CI can be difficult to maintain as it may lead
+  to conflicting updates of the consumer repo.
+
+Now that you understand the options, create or adjust a Git repository as needed.
+
+## Create the XCFramework and the Package.swift file
+
+> The following example assumes that the shared code of your KMP project is stored in the `shared` module.
+> If your project is structured differently, substitute "shared" in code and path examples with the name of your module.  
+>
+{type="tip"}
+
+To provide a Swift package, you can follow these steps:
+1. Set up publishing of an [XCFramework](multiplatform-build-native-binaries.md#build-xcframeworks). Add an `XCFramework`
+call to your iOS targets description in the `shared/build.gradle.kts` file:
+   ```kotlin
+   import org.jetbrains.kotlin.gradle.plugin.mpp.apple.XCFramework
+   
+   kotlin {
+      // other KMP targets
+      // ...
+      val xcf = XCFramework()
+      // Name of the module that will be imported in the consumer project
+      val xcframeworkName = "Shared"
+   
+      listOf(
+         iosX64(),
+         iosArm64(),
+         iosSimulatorArm64(),
+      ).forEach {
+           it.binaries.framework {
+              baseName = xcframeworkName
+              xcf.add(this)
+           }
+        }
+      //...
+   }
+   ```
+2. Sync the updated Gradle configuration.
+3. Run the Gradle task to create the framework:
+   
+   `./gradlew :shared:assembleSharedReleaseXCFramework`
+4. The resulting framework will be created as the `shared/build/XCFrameworks/release/shared.xcframework` folder in your project directory.
+5. Put the `shared.xcframework` folder in a ZIP archive and calculate the checksum for the resulting archive, for example:
+   
+   `swift package compute-checksum shared.xcframework.zip`
+6. Upload the ZIP file to the file storage of your choice.
+7. Create a `Package.swift` file with the following code:
+   ```Swift
+   // swift-tools-version:5.3
+   import PackageDescription
+    
+   let package = Package(
+      name: "shared",
+      platforms: [
+        .iOS(.v14),
+      ],
+      products: [
+         .library(name: "shared", targets: ["shared"])
+      ],
+      targets: [
+         .binaryTarget(
+            name: "shared",
+            url: "<link to the uploaded XCFramework ZIP file>",
+            checksum:"<checksum calculated for the ZIP file>")
+      ]
+   )
+   ```
+8. You can check if the package manifest is valid by running the following command next to the `Package.swift` file:
+
+    ```shell
+    swift package describe
```

Comment:
We should probably also recommend doing:
```
swift package reset && swift package show-dependencies --format json
```

This command will drop the `.build` directory and try to download the `.xcframework` that is pointed by `.binaryTarget`; it should help catch some simple configuration errors, such as incorrect URL, checksum or auth issues

Reply:
To fully check the correctness of the package, one can: 
1) Add `.testTarget(name: "sharedChech", dependencies: ["shared"])`
2) Add file to `Tests/sharedCheck/Check.swift` with one line `import shared`
3) Run `swift test`. It should pass if the framework and the package are correct
4) Remove what was added in 1) and 2)

But I'm not really sure whether it's a good idea to add it to this guide, because it's quite heavy.

Reply:
I think that would be an excellent "if you want to be thorough..." addition, not in the course of the main step-by-step, which is long already, but as another section at the end. One of those short and definitive recipes one spends hours tracking down on StackOverflow.